source/common/upstream/health_checker_impl.cc

```diff
@@ -279,15 +282,51 @@ void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::onInterval() {
 
 void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::onResetStream(Http::StreamResetReason,
                                                                         absl::string_view) {
+  request_in_flight_ = false;
+  ENVOY_CONN_LOG(debug, "connection/stream error health_flags={}", *client_,
+                 HostUtility::healthFlagsToString(*host_));
+  if (client_ && received_no_error_goaway_) {
+    client_->close();
```

Comment:
@alyssawilk this is the one place it might be weird to just use a `reuse_connection_` bool, because in the old behavior it looks like it would not close the connection if the stream was reset? That means i'd have to change the behavior a bit, but it also might be a small bug that it wasn't closing the connection before?

Reply:
I'll push the commit, we can always revert it

Reply:
yeah, closing if we can't reuse and the stream is reset seems fine, and it's behind the guard so even if someone gets surprised they can always revert the behavior

Reply:
some of this behavior change is not guarded. It looks like the same case came up when implementing GOAWAY handling in gRPC.

There is a slight behavior change if you have connection reuse disabled, and get a reset stream. Previously the connection would be reused! which seems wrong, but is an unguarded behavior change.

Reference PR: https://github.com/envoyproxy/envoy/pull/11325